Convert all doc files to ASCIIDoc

thatdocslady opened the pull request:
Fixes #531 

Converted all doc files to ASCIIDoc and incorporated some minor changes.

bexelbie commented:
> Open issues:
> TOC not live-rendered in GitHub but will appear in published docs (tested with asciidoctor). There is an ugly hack for this, but we need to decide if it's a priority.

I vote that this is not a priority.  Ultimately we want users to use the published docs and not to use the ones in the repository.  The docs in the repository should be like the code in the repository.  They need to be built to be useful.

> IDs have been auto-generated for each heading/section, we should examine and possibly change them to something more fitting for permalinks before we publish (it'll be harder to change later)

Yes.  Though I am less concerned about the actual ID (and url) than that it be stable.  So let's hardcode them auto-generated ones for now.  We have a a small user community, for now this can be a lower priority.

> There are references to these files from other repos i.e. vsm, should we wait and change the reference links there to the published URLs?

I believe for now, we shouldn't think about deep linking and should start with a reference to the head of the docs for that component.

> The files motivation.md and architecture.rst look outdated/obsolete and haven't been updated in a long time. Do we still need them or can we retire them?

+1 retire.

> Vagrantfile READMEs are also in RST, do we want to convert them? Will we be pulling them into the doc set? It's not a complex conversion but not sure where these end up

+1 for conversion to Asciidoc.  I don't believe we should mix formats.  However, I believe we ultimately need to look at not having them included in the docs.  They should become a standalone guide that gives you guidances for 30 seconds, 5 minutes, and 1 hour via links.

> Changelog.md - what is it used for? Should we convert it?

+1 for conversion to Asciidoc because of format similarity.  This should possibly be published like a Revision History.

> Side note:
> There are lots of deeply-nested bullet/numbered lists in the big docs (i.e. installing, using, readme), and it looks like they're threatening the serenity of file maintenance (not just in asciidoc). We should consider opening a separate issue for restructuring them into sections/sub-headings or some form of labeled lists, it'll make it easier to skim and find information.

Huge +1000

thatdocslady commented:
@LalatenduMohanty @Preeticp do you have anything to add/respond/counter for what @bexelbie commented?

I'm good with these suggestions, the only thing is like we discussed today I will replace deep-links with head-links in existing files, but keep the auto-generated asciidoc IDs for now (we will address them when we reach v3 publication, as they will be translated into the final html page names).

Preeticp commented:
- Two of the vagrantfile readmes (Kubernetes and Openshift) are used by CDK too and are included in the CDK Zip file, IIRC, @bexelbie , @LalatenduMohanty , can you confirm if this is right? Hope this conversion does not break anything in that. 
- Also, do we need to include Vagrant Registration repo (used by CDK) in this exercise? It only has one doc Readme.md. Currently we are targeting only ADB, VSM was already done. @LalatenduMohanty do we want to add any other repo within the ADB umbrella project?
- +1 to creating an issue for restructuring nested lists into sections/sub-headings, another idea discussed earlier was to have separate installation docs for the diff OS similar to the CDK docs, I think that would really help simplify a lot of these issues. We could also break the current Using doc on the basis of different developer use-cases. I think we could link this issue to issue #524.

thatdocslady commented:
Great feedback/suggestions, @Preeticp ! 

Here's the status update after the first review round:
- README files for plugins are now converted to asciidoc.
- No more deep-linking to subheadings in files.
- I added the task to restructure mega-crazy nested lists and/or split docs to the docs convergence suggestion box issue #524 
- I created a new issue #553 for cleanup tasks before we publish, because this has nothing to do with the docs convergence and there are things that we can't do now but must do before we can launch the doc site. 

Updated the PR description with the remaining open issues. @LalatenduMohanty please review and comment/approve as needed. I'm specifically looking for your opinion on the motivation.md and architecture.rst, @bexelbie already agreed that they should be retired but I think in this case I'd like another +1 for this before I delete.

brgnepal commented on CHANGELOG.adoc:
```diff
@@ -0,0 +1,223 @@
+= Revision History
```
Why Revision history? Not Changelog ? Thats more familiar to developers.

Preeticp replied:
AFAIU, only the heading inside the doc has been changed to reflect the contents more aptly. On the ADB landing page, it will continue to be CHANGELOG.adoc, unless the doc name is changed too.

LalatenduMohanty replied:
Yup, we should be good as the file name is `changelog.adoc`

thatdocslady replied:
@budhrg the reason for the internal name change is that these types of files can be used as prep files for release notes, because they indicate the changes that were made with each software release. This way, we can automate publishing this info from the file to the doc site in the future.

brgnepal commented on CHANGELOG.adoc:
```diff
@@ -0,0 +1,223 @@
+= Revision History
+:toc:
```
TOC not required here.

hferentschik replied:
TBH, I am not even sure whether this file should be converted at all. This is not really documentation per se. Keep this file format as simple as possible.

thatdocslady replied:
@budhrg if we publish this in the docs later on, it will be required. It doesn't interfere with anything and isn't live-rendered in github, so there's no reason to remove it.

@hferentschik I wrote to budhram in a different comment, these types of files can be used as a part of published release notes and it's better to have them publication-ready. The internal way of tagging bullet lists and titles in ADOC is almost identical to MD, if you compare the file contents of this file with its original MD file you'll see that it's still very simple :)

brgnepal commented on CHANGELOG.adoc:
```diff
@@ -0,0 +1,223 @@
+
+Release announcement: http://www.projectatomic.io/blog/2015/12/ADB-1-5/
+
+_Prior Releases were unannounced and mostly project-internal._
```
Rest of Changelog.adoc looks fine. :+1:

brgnepal commented on CONTRIBUTING.adoc:
```diff
@@ -0,0 +1,25 @@
+is considered 1 of the 2 ACKs. Therefore PRs from maintainers only
+require one additional ACK. By "2 ACKs" we mean that 2 maintainers must
+acknowledge that the change is a good one. The 2nd person to ACK the PR
+should merge the PR with a comment including their agreement.
```
:+1:

brgnepal commented on README.adoc:
```diff
@@ -0,0 +1,274 @@
+* Introduction to ADB from DevConf.cz 2016:
+https://www.youtube.com/watch?v=jxFw6qnGaRk
+* OpenShift in ADB Quickstart (video):
+https://www.youtube.com/watch?v=H58prwM3IbE
```
:+1:

brgnepal commented on components/centos/centos-docker-base-setup/README.adoc:
```diff
@@ -0,0 +1,37 @@
+----
+
+.  Proceed with using ADB as described in link:../../../docs/using.adoc[Using
+Atomic Developer Bundle].
```
:+1:

brgnepal commented on components/centos/centos-docker-base-setup/README.adoc:
```diff
@@ -0,0 +1,37 @@
+= Vagrantfile: CentOS Docker Base
+:toc:
```
TOC not required here.

thatdocslady replied:
@budhrg even in files that have one one 2heading, the TOC is useful because when we publish the docs it will be translated to a sidebar entry in the TOC tree.

brgnepal commented on components/rhel/README_CDK_ZIP.adoc:
```diff
@@ -0,0 +1,42 @@
+├── LICENSE
+├── plugins
+│   ├── vagrant-registration.gem
+│   ├── vagrant-registration-README.md
```
This .md file should be .adoc.

Preeticp replied:
Vagrant Registration and sshfs Readmes in both the line diagrams and the links mentioned on lines 38 and 42 need to be changed from .md to .adoc

brgnepal commented on docs/installing.adoc:
```diff
@@ -0,0 +1,313 @@
+https://docs.vagrantup.com/v2/installation/index.html[vagrantup.com].
+Testing has been done with version 1.7.4.
+
+* Fedora 22/23/Rawhide(24)
```
Could you pull the changes from upstream master? This doc has been updated with Fedora 23/24 support only.

thatdocslady replied:
@budhrg I couldn't cherry-pick the commit via git (don't ask, long story), so I looked into the original PR and manually edited the file to match the version numbers. Just a side note, this is why I asked for everyone to wait with editing content files while I was working, but luckily it wasn't a major change so now it'll be resolved once I update the PR.

Preeticp commented on README.adoc:
```diff
@@ -0,0 +1,274 @@
+** link:docs/cockpit.adoc[Using Cockpit with ADB]
+* link:docs/updating.adoc[Updating ADB]
+* link:docs/building.adoc[Building the Vagrant box] for Developers
+
```
Great, I see all the doc links have been updated!

thatdocslady replied:
Yes for now this is good, but ultimately they'll all need to be changed to the published doc URLs :)

hferentschik replied:
> Yes for now this is good, but ultimately they'll all need to be changed to the published doc URLs :)

Why? This should always stay relative links, right? You build a given version of the docs and then deploy it somewhere. The root is given by the deployment and all internal links within the docs should be relative. This also allows to deploy multiple versions of the doc in parallel. Having full URLs in there would be a maintenance nightmare.

LalatenduMohanty commented:
@thatdocslady @Preeticp +1 to @bexelbie comments including retiring the mentioned docs as they do not have valuable information for the community. 

@Preeticp Yes, vagrant-registration is only required for CDK. So you can skip that. README for CDK Vagrantfiles should be treated similar to ADB Vagrantfiles. Landrush is  a required Vagrant plugin for ADB. Not sure if you guys have similar plan for Landrush.

Preeticp commented on README.adoc:
```diff
@@ -0,0 +1,274 @@
+= Atomic Developer Bundle
+:toc:
+
```
Is the TOC a WIP? I dont see them rendered in any docs when I click View. 
In VSM they use a different syntax for TOC, refer https://raw.githubusercontent.com/projectatomic/vagrant-service-manager/master/README.adoc. Also, they follow the convention of placing the introductory paragraph before the TOC. I think both ways are good, but let us follow this consistently within the ADB umbrella Repos. WDYT?

hferentschik replied:
+1

BTW, don't you guys have a setup to let you edit and view the changes locally? It is a bit tedious to have to push to GitHub to see changes rendered.

thatdocslady replied:
@Preeticp I already mentioned this when I submitted the PR: the problem with the TOC being live-rendered here is apparently unrelated to the TOC in VSM, because the version of asciidoc changed and GitHub doesn't render TOCs anymore. I already made a comment about this and @bexelbie agreed that we shouldn't invest in it (I did a bunch of research and it looks like there are some really ugly hacks but no out-of-the-box way to do it now). 

@hferentschik I do use the ASCIIDoctor browser plugin as well as the Atom preview extension to test each file. This is how I know that the TOC will render correctly when we publish documentation to the final site. GitHub's TOC problem stems from its "safety" features where it disables some markup tags (other formats have their own problems, it's quite a fun ride). At this time, because we don't have a publication system, we cannot test the entire published doc set, but I do test everything locally before I push to the PR.

thatdocslady replied:
@hferentschik and I believe you meant to say "folks", not "guys".

hferentschik replied:
>  I already mentioned this when I submitted the PR: the problem with the TOC being live-rendered here is apparently unrelated to the TOC in VSM, 

How is it unrelated? Apparently TOC can be rendered on GitHub as one can see at vagrant-service-manager. So why can we not get them to render here? I think missing them on the rendered GitHub pages is a big loss.

> because the version of asciidoc changed and GitHub doesn't render TOCs anymore. 

So why does it render on vagrant-service-manager. I would love to understand this one better.

> I already made a comment about this and @bexelbie agreed that we shouldn't invest in it (I did a bunch of research and it looks like there are some really ugly hacks but no out-of-the-box way to do it now). 

So what are the hacks? Are there any GitHub internal issues somewhere?

Preeticp commented on docs/installing.adoc:
```diff
@@ -0,0 +1,313 @@
++
+Refer:
+link:../components/centos/centos-docker-base-setup/README.rst[README]
+** To configure a
```
Links to all readmes in this section-.rst>.adoc

Preeticp commented on docs/using.adoc:
```diff
@@ -0,0 +1,311 @@
+
+For a quick high level glimpse at ADB and how you can use it refer to
+the
+https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/README.rst[Readme].
```
.rst>.adoc

Preeticp commented on docs/using.adoc:
```diff
@@ -0,0 +1,311 @@
+execute commands and use the tools provided.
+
+You can use the
+https://github.com/projectatomic/adb-utils/blob/master/README.rst[sccli]
```
Considering, ADB utils will not be user facing, do we want to convert the docs in this repo too? This is the only place it is referenced and we can let it be .rst, I guess. 
Thoughts?

Preeticp commented on docs/using.adoc:
```diff
@@ -0,0 +1,311 @@
+Use the
+link:../components/centos/centos-k8s-singlenode-setup/Vagrantfile[Vagrantfile]
+and refer the corresponding
+link:../components/centos/centos-k8s-singlenode-setup/README.rst[README]
```
.rst>.adoc.
Maybe once this whole exercise is complete we can use https://wummel.github.io/linkchecker/ to check for broken links. I have not used it but it was highly recommended by one of the developers.

hferentschik commented on components/centos/centos-mesos-marathon-singlenode-setup/README.adoc:
```diff
@@ -0,0 +1,39 @@
+== Quickstart
+
+To setup a virtualization provider or to install Vagrant refer to the
+https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/docs/installing.adoc[ADB installation
```
Should that not be a relative link?

hferentschik commented on docs/installing.adoc:
```diff
@@ -0,0 +1,313 @@
+environmental concerns and is outside of the scope of this document.
+Both will work equally well in their default configuration. You may wish
+to read the section on file synchronization in the
+https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/docs/using.adoc[Using ADB documentation] when making this decision.
```
This should also be a relative link

brgnepal commented:
LGTM from my side for pushing conversion only changes. We can address the other issues in separately.

brgnepal commented on components/rhel/rhel-ose/README.adoc:
```diff
@@ -0,0 +1,90 @@
+= Vagrantfile: OpenShift Container Platform
```
:+1:
